src/PowerShellEditorServices/Services/PowerShell/Utility/PSCommandExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using System.Text;
+
+namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Utility
+{
+    internal static class PSCommandExtensions
+    {
+        public static PSCommand AddOutputCommand(this PSCommand psCommand)
+        {
+            return psCommand.MergePipelineResults()
+                .AddCommand("Out-Default", useLocalScope: true);
+        }
+
+        public static PSCommand AddDebugOutputCommand(this PSCommand psCommand)
+        {
+            return psCommand.MergePipelineResults()
+                .AddCommand("Out-String", useLocalScope: true)
+                .AddParameter("Stream");
+        }
+
+        public static PSCommand MergePipelineResults(this PSCommand psCommand)
+        {
+            Command lastCommand = psCommand.Commands[psCommand.Commands.Count - 1];
+            lastCommand.MergeMyResults(PipelineResultTypes.Error, PipelineResultTypes.Output);
+            lastCommand.MergeMyResults(PipelineResultTypes.Information, PipelineResultTypes.Output);
+            return psCommand;
+        }
+
+        public static string GetInvocationText(this PSCommand command)
```

Comment:
Might want to note this is explicitly for logging and *not* for passing to PowerShell to execute as a script.